src/main.cpp

```diff
@@ -3726,7 +3773,6 @@ bool ProcessNewBlock(CValidationState& state, const CChainParams& chainparams, c
       }
 
     LogPrint("thin", "Processing new block %s from peer %s (%d).\n", pblock->GetHash().ToString(), pfrom ? pfrom->addrName.c_str():"myself",pfrom ? pfrom->id: 0);
-    if (!IsInitialBlockDownload()) SendExpeditedBlock(*pblock,pfrom);
```

Comment:
Unfortunately it needs to be here to catch blocks coming from miners and from the relay network -- basically from the JSON interface, IIRC.  I don't like the double check either though.  I'm going to put this back in and then we can talk about what to do in a different PR.  That way we can get everything else in this PR merged.